include/kernel/of/fdt.h

```diff
@@ -21,19 +30,24 @@ struct fdt {
     uint32_t boot_cpuid_phys;
     uint32_t size_dt_strings;
     uint32_t size_dt_struct;
-};
+} KERNEL_PACKED KERNEL_ALIGNED(sizeof(void*));
 
 struct fdt_node {
     uint32_t token;
     char unit_name[];
-};
+} KERNEL_PACKED KERNEL_ALIGNED(4);
```

Comment:
Being aligned to `void*` in the first one is because the `memresv` table needs to be 4 byte aligned on 32-bit systems and 8 byte aligned on 64-bit systems (which forces the header to be similarly aligned)

Reply:
Reason is to make it explicit what you are aligning to. 

By saying `sizeof(uint32_t)`, you're saying "I want this aligned to the size of a `uint32_t`"

By saying `4`, you could have just arbitrarily picked the number because it was nice or something.

Reply:
I am making it explicit, I need it aligned to exactly `4` independently of `uint32_t` which just happens to have size `4` in this case

Reply:
I know `sizeof(uint32_t)` will always be `4`, but so will `ceil(M_PI)`, and all we really care about is a 4 byte alignment, not the `uint32_t` type

Reply:
alright sure